packages/taquito-michelson-encoder/src/tokens/token.ts

```diff
@@ -103,6 +103,13 @@ export abstract class Token {
 
   public abstract EncodeObject(args: any, semantics?: SemanticEncoding): any;
 
+  /**
+   *
+   * @description checks the value of what is passed validating the same way as EncodeObject and throws error if necessary
+   * @param arg same as EncodeObject
```

Comment:
`arg same as EncodeObject` ->
Not exactly, there are discrepancies between EncodeObject, and this is why we need a separate TypecheckValue method to handle those case